structure/heap/heap.go

```diff
@@ -0,0 +1,79 @@
+package heap
+
+// The Interface type describes the requirements
+// for a type using the routines in this package.
+type Interface[T any] interface {
+	Less(o T) bool
+}
+
+// Heap heap implementation using generic.
+type Heap[T Interface[T]] struct {
+	heaps []T
+}
```

Comment:
```suggestion
type Heap[T constraints.Ordered] struct {
	heaps []T
}
```

Reply:
If we need this Interface,  this modificarion should not be made.